File: SecurityManagementApp/src/main/java/com/boulevardsecurity/securitymanagementapp/service/LigneCotisationService.java
// src/main/java/com/boulevardsecurity/securitymanagementapp/service/LigneCotisationService.java
package com.boulevardsecurity.securitymanagementapp.service;

import com.boulevardsecurity.securitymanagementapp.dto.LigneCotisationCreationDto;
import com.boulevardsecurity.securitymanagementapp.dto.LigneCotisationDto;

import java.util.List;
import java.util.Optional;

public interface LigneCotisationService {
    List<LigneCotisationDto> getAll();
    Optional<LigneCotisationDto> getById(Long id);
    List<LigneCotisationDto> getByFicheDePaieId(Long ficheId);
    LigneCotisationDto create(LigneCotisationCreationDto dto);
    LigneCotisationDto update(Long id, LigneCotisationCreationDto dto);
    void delete(Long id);
}

File: security-management-frontend/src/services/LigneCotisationService.js
import api from "./api"; // axios.create({ baseURL: "http://localhost:8080/api" })

const LigneCotisationService = {
    getAll: () =>
        api.get("/lignes-cotisation"),
    getById: id =>
        api.get(`/lignes-cotisation/${id}`),
    getByFiche: ficheId =>
        api.get(`/lignes-cotisation/fiche/${ficheId}`),
    create: dto =>
        api.post("/lignes-cotisation", dto),
    update: (id, dto) =>
        api.put(`/lignes-cotisation/${id}`, dto),
    delete: id =>
        api.delete(`/lignes-cotisation/${id}`)
};

export default LigneCotisationService;
